pkg/api/http/metadata.go

```diff
@@ -93,13 +93,15 @@ func (a *api) onGetMetadata() http.HandlerFunc {
 
 				// Copy the subscriptions into a custom struct
 				if len(out.GetSubscriptions()) > 0 {
-					subs := make([]metadataResponsePubsubSubscription, len(out.GetSubscriptions()))
+					subs := make([]MetadataResponsePubsubSubscription, len(out.GetSubscriptions()))
```

Comment:
had to put this as exportable to be able to use in the integration tests. as mentioned in the TODO below it would be better/cleaner to change these to use proto but then one of the tests was failing as the return value was returning .Rules.Rules that was not expected in the tests, not a big deal but not sure how people might be using this.

Reply:
Fair.

Reply:
Please keep this type un-exported- where needed we copy the type across into the integration test as and where needed.

This API is not particularly "designed" out like the other first class APIs and exporting the types makes things worse. Long term, indeed these should be "promoted"  over to proto.